torch/ao/quantization/fx/prepare.py

```diff
@@ -1235,8 +1228,17 @@ def insert_observers_for_model(
                 not node.op == 'output'
             )
 
-            is_supported_by_backend = is_pattern_dtype_config_supported_by_backend(
-                pattern, matched_node_pattern, node_name_to_target_dtype, backend_config)
+            # Check whether this pattern's dtype is supported by the backend
+            if pattern is not None and matched_node_pattern is not None and backend_config is not None:
+                supported_dtype_config = _get_supported_pattern_dtype_config(
+                    pattern, matched_node_pattern, node_name_to_target_dtype, backend_config)
+                is_supported_by_backend = supported_dtype_config is not None
+            else:
+                supported_dtype_config = None
+                is_supported_by_backend = True
+
+            if qconfig is not None and supported_dtype_config is not None:
+                _validate_qconfig_against_dtype_config(qconfig, supported_dtype_config, pattern)
```

Comment:
I feel we should do this in `is_pattern_dtype_config_supported_by_backend` function directly, why do we do this in a separate function?

Reply:
Good point. I'll do the validation in `is_pattern_dtype_config_supported_by_backend`.

Reply:
Ok, I ended up keeping the validation code in utils to keep the body of `is_pattern_dtype_config_supported_by_backend` small, but I still call the validation code in there